[PATCH] ar71xx: add anonabox-pro target [v6]

augustgermar opened the pull request:
[PATCH] Add support for anonabox_pro hardware [v6]

Updated version with suggested changes from previous PR: #110
link:  https://github.com/openwrt/openwrt/pull/110

Chipset is a 650mhz atheros with 128mb ram.
Variation of qca9531 SoC with two Ethernet ports and a 802.11n 2.4 GHz radio
Includes usb configuration and led light
Includes minor changes to patch file for auto compiler from previous version.
Prompt for y/n during compile removed.
Uses current kernel version for ar97xx 
Suggested changes from IRC implemented, including removal of patch file from previous version to be 'cleaner'
Tested working on hardware.
Comments Welcome.
I will be in IRC also if there are any questions.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
have the right to submit it under the open source license
indicated in the file; or

(b) The contribution is based upon previous work that, to the best
of my knowledge, is covered under an appropriate open source
license and I have the right under that license to submit that
work with modifications, whether created in whole or in part
by me, under the same open source license (unless I am
permitted to submit under a different license), as indicated
in the file; or

(c) The contribution was provided directly to me by some other
person who certified (a), (b) or (c) and I have not modified
it.

(d) I understand and agree that this project and the contribution
are public and that a record of the contribution (including all
personal information I submit with it, including my sign-off) is
maintained indefinitely and may be redistributed consistent with
this project or the open source license(s) involved.

Signed-off-by: August Germar august@anonabox.com

augustgermar committed:
add anonabox-pro target

augustgermar committed:
add signed off by

specify chipset for organization in Board name, add developers origin

augustgermar committed:
add gnu / free software text

diizzyy commented on target/linux/ar71xx/base-files/etc/board.d/02_network:
```diff
@@ -530,6 +530,10 @@ wndr3700)
 	ucidef_add_switch_port_attr "switch0" 5 led 2
 	;;
 
+anonabox-pro)
```
Is this really needed given L537? (I'm not sure so don't take my word for it)

augustgermar replied:
You are right, this was verified by another person in IRC. Thank you. I will make the change.

diizzyy commented on target/linux/ar71xx/generic/profiles/anonabox.mk:
```diff
@@ -0,0 +1,18 @@
+
+define Profile/ANONABOX_PRO
+	NAME:=AnonaBox Pro board
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb-uhci \
```
Less is more, this is most likely not going to be accepted as is. Have a look at the minimal config or similar ones.

diizzyy reviewed (commented):
Just some minor things...

augustgermar committed:
reduce list of packages to reflect suggestion

Used 00-default.mk and 01-minimal.mk as reference.

augustgermar committed:
Update 02_network as per suggestion

Removed Catchall from line 537. Thanks for the suggestion!

augustgermar committed:
update as per suggestion from IRC

There is no need to make any changes to this file to support new target, this commit should revert it to current trunk.

augustgermar reviewed (commented):
only 9 files are to be changed now instead of 10 because I removed 02_network changes.

augustgermar commented:
Thanks everyone for your feedback! I'll be in IRC if you want to chat about anything. SO far I have implemented all suggestions and they improved the code considerably. If there is any other comments of feedback let me know. Otherwise the patch will sit here waiting to be merged.

wigyori commented:
Hi @augustgermar - this is just nitpicking about the commit messages, as neither of the commit messages are in the requested form. I'll pull these together into my tree and let You review it.

wigyori commented:
@augustgermar, would you mind to take a look at https://github.com/wigyori/openwrt/commit/aac4a5b6e9b52383df23e3b589f2458e7a08849d , thanks.

augustgermar commented:
Yes, looks good to me, thanks for the help with formatting and for your
comments as I am still learning.

On Sat, Jan 7, 2017 at 7:02 AM, Zoltan Herpai <notifications@github.com>
wrote:

> @augustgermar <https://github.com/augustgermar>, would you mind to take a
> look at wigyori@aac4a5b
> <https://github.com/wigyori/openwrt/commit/aac4a5b6e9b52383df23e3b589f2458e7a08849d>
> , thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openwrt/openwrt/pull/249#issuecomment-271088591>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AUR_w1xxsPcUo_mrTBVsIHB780ro0JUhks5rP6kWgaJpZM4K-F1M>
> .
>

wigyori commented:
No probs, that's great you're working on this :)

wigyori commented:
This was refreshed in https://github.com/openwrt/openwrt/pull/320 and have been merged.
